### DISCUSSION TITLE
Building on multiple operating systems

### DISCUSSION THREAD

**Discussion by alinjie** [goreleaser/goreleaser#2862] [Q&A] [answered]
Hi!

I have project that has to be built on two different operating systems, specifically Windows and macOS, and I would like to use GoReleaser to build, sign and publish the artifacts.

My initial thought was to do create a GitLab CI pipeline that does something like this:
1. Run `goreleaser build --id windows` in one job
2. Run `goreleaser build --id darwin` in another job
3. Gather the output from job 1 and 2 and publish.

As the `dist` folder has to be empty in order to release, I guess this workflow cannot be achieved but I'm having difficulties finding an alternate method. I was thinking I could have two different GoReleaser configurations and build/release on both operating systems separatly, but that would result in two releases in GitLab I guess.

Here is my current config:
```yaml
project_name: my-project
signs:
  - ids:
    - windows
    cmd: powershell
    args:
    - -File scripts/SignWindowsBinary.ps1 -Path {{ .ArtifactPath }}
    artifacts: all
builds:
  - id: darwin
    main: main_darwin.go main.go
    ldflags:
      - -X 'main.Version={{ .Tag }}'
    hooks:
      pre:
        - go mod download
    goos:
      - darwin
    goarch:
      - arm64
      - amd64
  - id: windows
    main: ./
    ldflags:
      - -X 'main.Version={{ .Tag }}'
    hooks:
      pre:
        - go mod download
    goos:
      - windows
    goarch:
      - "386"
      - amd64
```

**To summarize:**
I want to run the `windows` build and sign process on Windows, run the `darwin` build on macOS and publish the resulting artifacts as a single GitLab release.

Apologies if I'm missing something fundamental here.

**Comment by caarlos0** [comment, marked as answer]
You'll likely need goreleaser-pro and 2 goreleaser config files - https://carlosbecker.com/posts/goreleaser-prebuilt/

OR, use a CGO docker image, which I'm assuming is what is going on

**Reply by alinjie** [reply]
Thank you!
